src/System.IO.Pipelines/tests/StreamPipeReaderTests.cs

```diff
@@ -568,6 +576,25 @@ private static object[] CreateRead(int bytesInBuffer, int bufferSize, int minimu
             return new object[] { bytesInBuffer, bufferSize, minimumReadSize, readSizes };
         }
 
+        private class ThrowsOperationCanceledExceptionStream : ReadOnlyStream
+        {
+            public override int Read(byte[] buffer, int offset, int count)
+            {
+                throw new OperationCanceledException();
+            }
+
+            public override Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+            {
+                throw new OperationCanceledException();
+            }
+#if netcoreapp
+            public override ValueTask<int> ReadAsync(Memory<byte> buffer, CancellationToken cancellationToken = default)
+            {
+                throw new OperationCanceledException();
+            }
```

Comment:
It doesn't actually matter given how the code is written, but generally OCEs would come out stored in the returned task rather than propagating out synchronously.